src/azstoragetorch/_client.py

```diff
@@ -88,6 +91,7 @@ def get_blob_size(self) -> int:
         return self._blob_properties.size
 
     def download(self, offset: int = 0, length: Optional[int] = None) -> bytes:
+        _LOGGER.debug("download")
```

Comment:
Then for the blob client, what we will probably need to do here:
1. Refactor the `_blob_properties` to not immediately try to grab the properties if we access it. Instead at the very beginning of the `download()` method, determine if the size is cached.
2. If the size is not cached, attempt a `_download_with_retries` using the length or partition threshold (whichever is less). As part of this download we will want to parse the size and etag from the headers similar to what is currently being done in `BlobIO`.
3. Continue on with the download logic if there is more content to retrieve and account for the already retrieved amount.